src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/CertificateGenerator.cs

```diff
@@ -216,22 +216,28 @@ public TimeSpan ValidityPeriod
             }
         }
 
-        public X509CertificateContainer CreateMachineCertificate(params string[] subjects)
+        public X509CertificateContainer CreateMachineCertificate(CertificateCreationSettings creationSettings = null)
         {
             EnsureInitialized();
-            return CreateCertificate(false, true, _authorityCertificate.InternalCertificate , subjects);
+            return CreateCertificate(false, true, _authorityCertificate.InternalCertificate , creationSettings);
         }
 
-        public X509CertificateContainer CreateUserCertificate(params string[] subjects)
+        public X509CertificateContainer CreateUserCertificate(CertificateCreationSettings creationSettings = null)
         {
             EnsureInitialized();
-            return CreateCertificate(false, false, _authorityCertificate.InternalCertificate, subjects);
+            return CreateCertificate(false, false, _authorityCertificate.InternalCertificate, creationSettings);
         }
 
         // Only the ctor should be calling with isAuthority = true
         // if isAuthority, value for isMachineCert doesn't matter
-        private X509CertificateContainer CreateCertificate(bool isAuthority, bool isMachineCert, X509Certificate signingCertificate, params string[] subjects)
+        private X509CertificateContainer CreateCertificate(bool isAuthority, bool isMachineCert, X509Certificate signingCertificate, CertificateCreationSettings certificateCreationSettings)
         {
+            if (certificateCreationSettings == null)
+            {
+                certificateCreationSettings = new CertificateCreationSettings();
+            }
```

Comment:
I think the only situation where `certificateCreationSettings == null` should be allowed is where `isAuthority == true`. We should always throw here otherwise.